engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -900,62 +900,56 @@ namespace dmGameSystem
 
     struct DispatchContext
     {
-        PhysicsContext* m_PhysicsContext;
-        bool m_Success;
-        dmGameObject::HCollection m_Collection;
-        CollisionWorld* m_World;
+        PhysicsContext*         m_PhysicsContext;
+        dmGameObject::HRegister m_Register;
+        uint32_t                m_ComponentTypeIndex;
+        bool                    m_Success;
     };
 
-    void DispatchCallback(dmMessage::Message *message, void* user_ptr)
+    void DispatchCallback(dmMessage::Message* message, void* user_ptr)
     {
         DispatchContext* context = (DispatchContext*)user_ptr;
-        if (message->m_Descriptor != 0)
+
+        if (message->m_Descriptor == 0)
+            return;
+
+        dmDDF::Descriptor* descriptor = (dmDDF::Descriptor*)message->m_Descriptor;
+        if (descriptor == dmPhysicsDDF::RequestRayCast::m_DDFDescriptor)
         {
-            dmDDF::Descriptor* descriptor = (dmDDF::Descriptor*)message->m_Descriptor;
-            if (descriptor == dmPhysicsDDF::RequestRayCast::m_DDFDescriptor)
+            dmPhysicsDDF::RequestRayCast* ddf = (dmPhysicsDDF::RequestRayCast*)message->m_Data;
+
+            dmhash_t coll_name_hash = dmMessage::GetSocketNameHash(message->m_Sender.m_Socket);
+
+            // Target collection which can be different than we are updating for.
+            dmGameObject::HCollection collection = dmGameObject::GetCollectionByHash(context->m_Register, coll_name_hash);
+            if (!collection) // if the collection has been removed
+                return;
+
+            // NOTE! The collision world for the target collection is looked up using this worlds component index
+            //       which is assumed to be the same as in the target collection.
+            uint32_t component_type_index = context->m_ComponentTypeIndex;
+            CollisionWorld* world = (CollisionWorld*) dmGameObject::GetWorld(collection, component_type_index);
```

Comment:
We no longer need the go instance at all, we just need the collection to find the CollisionWorld*.